presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -183,13 +183,21 @@ public ActualProperties visitEnforceSingleRow(EnforceSingleRowNode node, List<Ac
         public ActualProperties visitAssignUniqueId(AssignUniqueId node, List<ActualProperties> inputProperties)
         {
             ActualProperties properties = Iterables.getOnlyElement(inputProperties);
+
+            ImmutableList.Builder<LocalProperty<Symbol>> localPropertiesBuilder = ImmutableList.builder();
+            localPropertiesBuilder.addAll(properties.getLocalProperties());
+            localPropertiesBuilder.add(new GroupingProperty<>(ImmutableSet.of(node.getIdColumn())));
```

Comment:
I think we can make stronger statement here that we are grouped on all columns, no?

Reply:
@sopel39 Karol, I think you are correct. Thanks for pointing it out. It might be the final piece I was missing.

We can say that the data is grouped on `{unique + subset-of-source-symbols}` for all subsets of the source symbols. This property will survive the join and will allow streaming aggregation without any hashing (as you suggested earlier). I'll implement `StreamingAggregationOperator` similar to `AggregationOperator` and drop changes to `HashAggregationOperator`.